include/gui_templates.h

```diff
@@ -31,37 +31,26 @@
 #include <QFont>
 #include <QDesktopWidget>
 
+// TODO: cleanup for qt6
+#if (QT_VERSION < QT_VERSION_CHECK(6,0,0))
+	#include <QScreen>
+#else
+	#include <QGuiApplication>
+#endif
+
 namespace lmms
 {
 
 
 // return DPI-independent font-size - font with returned font-size has always
 // the same size in pixels
-template<int SIZE>
-inline QFont pointSize( QFont _f )
+inline QFont pointSize(QFont _f, float SIZE)
 {
-	static const float DPI = 96;
-#ifdef LMMS_BUILD_WIN32
-	_f.setPointSizeF( ((float) SIZE+0.5f) * DPI /
-			QApplication::desktop()->logicalDpiY() );
-#else
-	_f.setPointSizeF( (float) SIZE * DPI /
-			QApplication::desktop()->logicalDpiY() );
-#endif
-	return( _f );
-}
-
+	// to calculate DPI of a screen to make it HiDPI ready
+	qreal devicePixelRatio = QGuiApplication::primaryScreen()->devicePixelRatio();
+    qreal scaleFactor = devicePixelRatio > 1.0 ? devicePixelRatio : 1.0; // Ensure scaleFactor is at least 1.0
```

Comment:
i think we should use `std::max(1.0, devicePixelRatio)` for better readability

Reply:
Why didn't i think of this?

Reply:
done with https://github.com/LMMS/lmms/pull/7159/commits/296583e8d3cd676b171b86ea7135663d11acf0a7